sdk/test-utils/recorder-new/MIGRATION.md

```diff
@@ -0,0 +1,160 @@
+# Migration Guide
+
+This document outlines key differences between the legacy recorder and the new Unified Recorder client.
```

Comment:
Please mention the docker prerequisite

Reply:
It's mentioned, but I'll pull it up: https://github.com/Azure/azure-sdk-for-js/blob/4264ca391c83abbd46b9fff43397686b4a776d02/sdk/test-utils/recorder-new/MIGRATION.md?plain=1#L19